ca/transport.go

```diff
@@ -135,6 +135,36 @@ func (c *MutableTLSCreds) LoadNewTLSConfig(newConfig *tls.Config) error {
 	return nil
 }
 
+// UpdateCAs updates the root CAs and client CAs of the existing TLS config in place
+func (c *MutableTLSCreds) UpdateCAs(rootCAs, clientCAs *x509.CertPool) {
+	c.Lock()
+	defer c.Unlock()
+	// TODO(cyli): on upgrade to go 1.8.x, use c.config.Clone() instead
```

Comment:
`Clone` exists in 1.7.

Would you mind switching to it before merging?

AFAIK there's no reason we need to support Go 1.6. @vdemeester do you know if there is?